_includes/date.html

```diff
@@ -0,0 +1,42 @@
+{% if include.lang and include.lang != empty %}
+  {% assign plang = include.lang %}
+{% else %}
+  {% assign plang = page.lang | default: site.lang | default: "en" %}
+{% endif %}
+{% assign splang = plang | truncate: 2, "" %}
+{% if plang != "en" %}
+  {% assign dateformat = site.data.i18n.[plang].date.format %}
+  {% unless dateformat %}
+    {% assign dateformat = site.data.i18n.[splang].date.format %}
+  {% endunless %}
+  {% unless dateformat %}
+    {% assign dateformat = "%b %-d, %Y" %}
+  {% endunless %}
+  {% comment %} The dateformat is determined based on the language. Now replace required variables (for now we ignore time based variables (%p, %P)) {% endcomment %}
+  {% assign num_month = include.date | date: "%-m" | minus: 1 %}
+  {% assign num_day = include.date | date: "%w" | plus: 0%}
+  {% assign i18n_dateformat = dateformat %} 
+  {% if site.data.i18n.[plang].date.abbr_day_names[num_day] %}
+    {% assign i18n_dateformat = i18n_dateformat | replace:"%a", site.data.i18n.[plang].date.abbr_day_names[num_day] %}
+  {% elsif site.data.i18n.[splang].date.abbr_day_names[num_day] %}
+    {% assign i18n_dateformat = i18n_dateformat | replace:"%a", site.data.i18n.[splang].date.abbr_day_names[num_day] %}
+  {% endif %}
```

Comment:
How about this? It can be done similarly for `day_name`, `abbr_month_name`, and `month_name`.

```suggestion
  {% assign abbr_day_name = site.data.i18n.[plang].date.abbr_day_names[num_day] | default: site.data.i18n.[splang].date.abbr_day_names[num_day] %}
  {% if abbr_day_name %}
    {% assign i18n_dateformat = i18n_dateformat | replace: "%a", abbr_day_name %}
  {% endif %}
```